applications/serial_lte_modem/src/slm_at_tcpip.c

```diff
@@ -197,16 +201,40 @@ static int do_socket_open(uint8_t type, uint8_t role, int sec_tag)
 		goto error_exit;
 	}
 
-	if (sec_tag != INVALID_SEC_TAG) {
-		sec_tag_t sec_tag_list[1] = { sec_tag };
+	if (client.sec_tag != INVALID_SEC_TAG) {
+		sec_tag_t sec_tag_list[1] = { client.sec_tag };
+#if defined(CONFIG_SLM_NATIVE_TLS)
+		int verify;
 
+		ret = slm_tls_loadcrdl(client.sec_tag);
+		if (ret < 0) {
+			LOG_ERR("Fail to load credential: %d", ret);
+			return ret;
+		}
+
+		/* Set up default TLS peer verification */
+		if (role == AT_SOCKET_ROLE_SERVER) {
+			verify = TLS_PEER_VERIFY_NONE;
+		} else {
+			verify = TLS_PEER_VERIFY_REQUIRED;
+		}
```

Comment:
hardcoded decision for peer_verify

Reply:
Additional "level" parameter is needed in XSOCKETOPT to set proper peer_verify option. Currently the peer_verify is hardcoded as default mbedTLS setting.